core/templates/pages/contributor-dashboard-admin-page/services/contributor-dashboard-admin-backend-api.service.spec.ts

```diff
@@ -0,0 +1,351 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ContributorDashboardAdminBackendApiService.
+ */
+
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+
+import { ContributorDashboardAdminBackendApiService } from './contributor-dashboard-admin-backend-api.service';
+import { CsrfTokenService } from 'services/csrf-token.service';
+
+describe('Contributor dashboard admin backend api service', () => {
+  let cdabas: ContributorDashboardAdminBackendApiService;
+  let httpTestingController: HttpTestingController;
+  let csrfService: CsrfTokenService = null;
+  let successHandler = null;
+  let failHandler = null;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+
+    cdabas = TestBed.get(ContributorDashboardAdminBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
+    csrfService = TestBed.get(CsrfTokenService);
+    successHandler = jasmine.createSpy('success');
+    failHandler = jasmine.createSpy('fail');
+
+    spyOn(csrfService, 'getTokenAsync').and.callFake(async() => {
+      return Promise.resolve('sample-csrf-token');
+    });
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should add contribution rights to the user given the' +
+    'name when calling addContributionReviewerAsync', fakeAsync(() => {
+    let category = 'voiceover';
+    let languageCode = 'en';
+    let username = 'validUser';
+    let payload = {
+      category: category,
+      username: username,
+      language_code: languageCode
+    };
```

Comment:
Nit: the variables that are not reassigned can be declared as `const`. Here and elsewhere.